adala/runtimes/_openai.py

```diff
@@ -156,19 +162,21 @@ class OpenAIChatRuntime(Runtime):
         max_tokens: Maximum number of tokens to generate. Defaults to 1000.
     """
 
+    class Config:
+        arbitrary_types_allowed = True  # for @computed_field
+
     openai_model: str = Field(alias="model")
     openai_api_key: Optional[str] = Field(
         default=os.getenv("OPENAI_API_KEY"), alias="api_key"
     )
     max_tokens: Optional[int] = 1000
     splitter: Optional[str] = None
 
-    _client: OpenAI = None
+    @computed_field
+    def _client(self) -> OpenAI:
```

Comment:
why do we need computed_field here but create a client on the fly in the Async version? let's use the unique approach if possible

Reply:
because the sync runtime actually uses its `_client`, but the async runtime does not, it directly sends post requests in `async_create_completion`. Even if it did use the python client lib instead of requests, it wouldn't make sense for the class to carry a single client instance because it would create them on demand to use them concurrently

Reply:
@niklub if you feel strongly about this I'll change it, but I think it's a meaningful difference that the async openai runtime doesn't need a `self._client`